authz/providers/azure/rbac/rbac.go

```diff
@@ -224,9 +225,11 @@ func (a *AccessInfo) RefreshToken(ctx context.Context) error {
 
 		// Set the authorization headers for future requests
 		a.headers.Set("Authorization", fmt.Sprintf("Bearer %s", resp.Token))
-		expIn := time.Duration(resp.Expires) * time.Second
-		a.expiresAt = time.Now().Add(expIn - expiryDelta)
-		klog.Infof("Token refreshed successfully on %s. Expire at:%s", time.Now(), a.expiresAt)
+
+		// Use ExpiresOn to set the expiration time
+		expOn := time.Unix(int64(resp.ExpiresOn), 0)
+		a.expiresAt = expOn.Add(-tokenExpiryDelta)
```

Comment:
Will the tokenProvider.Acquire guarantee to refresh the token or it's possible to load from a cache? In the latter case, will there be chance that the `resp.ExpiresOn` is less than 5 mins from time.Now(), which causes the a.expiresAt < time.Now() ? What's the impact of seeing this, like will that cause the RefreshToken being call repeatedly and eventualy cause busy loop?

Reply:
So, from the tokenProvider.Acquire point of view, it's always going to get a token with at-least 1 hour of expiry left
As the identity controller always refreshes the token 1 hour before expiry - 
https://github.com/azure-core/ClusterConfigurationAgent/blob/9e24b19ffdc7f71f53cb4fb86cb3d7d830be6856/Utils/TokenProviders/ManagedIdentityTokenProvider.go#L78

So, we should never hit a scenario where the ExpiresOn is less than 5mins from time.Now()

Reply:
For context, in Arc we have on our own `msi-adapter` component that mimic IMDS on the edge. Like @saisankargochhayat it just fetch a "cached" token from another component "identity-controller" that manage getting token and save it as k8s secret via a CR. The code that @saisankargochhayat linked is where `msi-adapter` decide to trigger a token rotation of re-used the cache secret within the 1H expiration.